lib/install/management/network.go

```diff
@@ -95,26 +99,47 @@ func (d *Dispatcher) removeNetwork(conf *config.VirtualContainerHostConfigSpec)
 		return nil
 	}
 
-	name := conf.Name
-	if network, err := d.session.Finder.Network(d.ctx, name); err != nil || network == nil {
-		log.Infof("Didn't find network %q", name)
-		log.Debugf("Didn't find network for %s", err)
+	log.Debugf("Remove bridge network based on %s", conf.ExecutorConfig.Networks["bridge"].Network.ID)
+
+	name := conf.ExecutorConfig.Networks["bridge"].Network.ID
+	moref := new(types.ManagedObjectReference)
+	ok := moref.FromString(name)
+	if !ok {
+		return fmt.Errorf("Unable to delete port group - failed to get moref from: %q", name)
+	}
+
+	net, err := d.session.Finder.ObjectReference(d.ctx, *moref)
+	if err != nil {
+		return fmt.Errorf("Unable to delete port group - failed to find network from: %q", name)
+	}
+	log.Debugf("Delete bridge network: %s", net)
+
+	netw, ok := net.(*object.Network)
+	if !ok {
+		log.Errorf("Failed to find network %q: %s", moref, err)
+		return err
+	}
+
+	if network, err := d.session.Finder.Network(d.ctx, netw.InventoryPath); err != nil || network == nil {
+		log.Infof("Unable to delete network - failed to find %q", netw.InventoryPath)
+		log.Debugf("Failed to find network to delete: %s", err)
 		return nil
 	}
 
-	log.Infof("Removing Portgroup %q", name)
+	pgName := netw.Name()
+	log.Infof("Removing Portgroup %q", pgName)
 	hostNetSystem, err := d.session.Host.ConfigManager().NetworkSystem(d.ctx)
 	if err != nil {
 		return err
 	}
 
-	err = hostNetSystem.RemovePortGroup(d.ctx, name)
+	err = hostNetSystem.RemovePortGroup(d.ctx, pgName)
 	if err != nil {
 		return err
 	}
 
-	log.Infof("Removing VirtualSwitch %q", name)
-	err = hostNetSystem.RemoveVirtualSwitch(d.ctx, name)
+	log.Infof("Removing VirtualSwitch %q", pgName)
```

Comment:
So, you don't want to exit here. You should just log that you could not delete something and proceed to the next item.